docs/site/Validation-REST-layer.md

```diff
@@ -175,3 +175,47 @@ of 10 digits separated by `-` after the 3rd and 6th digits.
 ```
 
 {% include tip.html content="RegExp can be converted into a string with `.source` to avoid escaping backslashes" %}
+
+## Customize validation errors
+
+Since the error is being caught at the REST layer, the best way to customize the
+errors is to
+[customize the sequence of actions](https://loopback.io/doc/en/lb4/Sequence.html#customizing-sequence-actions)
+using a
+[Provider](https://loopback.io/doc/en/lb4/apidocs.context.provider.html).
+
+To do so, create a Provider class in `src/providers` folder that implements
+`Provider<Reject>`. Using the
+[validation-app](https://github.com/strongloop/loopback-next/tree/master/examples/validation-app)
+as an example, here is how the Provider class look like:
+
+{% include code-caption.html content="/src/providers/my-validation-error-provider.ts" %}
+
+```ts
+import {HandlerContext, HttpErrors, RejectProvider} from '@loopback/rest';
+
+export class MyValidationErrorProvider extends RejectProvider {
+  // ...
+
+  action({request, response}: HandlerContext, error: Error) {
+    // handle the error and send back the error response
+    // "response" is an Express Response object
+    response.setHeader('Content-Type', 'application/json');
+    const httpError = <HttpErrors.HttpError>error;
+    if (request.url === '/coffee-shops') {
+      // if this is a validation error
+      if (httpError.statusCode === 422) {
+        const newError = {
+          message: 'My customized validation error message',
+          code: 'VALIDATION_FAILED',
+          resolution: 'Contact your admin for troubleshooting.',
+        };
+
+        // you can change the status code here too
+        response.status(422).send(JSON.stringify(newError));
+      }
```

Comment:
Missing `return`?

At the moment, you send the custom response and also call the default reject provider. Is that intended?

Fortunately, our error handler detects the situation when the response has been already sent and becomes no-op, so this isn't going to crash the app, but still.

An important side effect to be aware of: `RejectProvider` is handling error logging too. Depending on how you write your Reject provider, the error may or may not be logged for custom error path. (Your current solution will log the error because it always call `super.action`).

Since your example is interested only in changing the HTTP response,  maybe it would be better to change `RejectProvider` to receive the implementation of `writeErrorToResponse` via dependency injection too, so that code like yours can tweak only the part creating the HTTP response?

https://github.com/strongloop/loopback-next/blob/29e2076bff3cce7ca4445bde720f4cf9f1dae0e9/packages/rest/src/providers/reject.provider.ts#L41